src/epochs.lua

```diff
@@ -821,26 +800,25 @@ function convertPrescribedEpochToDistributedEpoch(epoch, currentTimestamp, distr
 	}
 end
 
-function epochs.scheduleNextEpochsPrune(timestamp)
-	NextEpochsPruneTimestamp = math.min(NextEpochsPruneTimestamp or timestamp, timestamp)
-end
-
-function epochs.nextEpochsPruneTimestamp()
-	return NextEpochsPruneTimestamp
-end
-
+--- Gets the reward rate for an epoch. The reward rate is the percentage of the protocol balance that is distributed to the gateways and observers.
+--- For the first year, the reward rate is 0.1% of the protocol balance.
+--- After the first year, the reward rate decays linearly to 0.05% of the protocol balance after 1.5 years.
 ---@param epochIndex number
 ---@returns number
 function epochs.getRewardRateForEpoch(epochIndex)
-	if epochIndex <= constants.rewardDecayStartEpoch then
-		return constants.minimumRewardRate
-	elseif epochIndex <= constants.rewardDecayLastEpoch then
-		local totalDecayPeriod = (constants.rewardDecayLastEpoch - constants.rewardDecayStartEpoch) + 1
-		local epochsAlreadyDecayed = (epochIndex - constants.rewardDecayStartEpoch)
-		local decayRatePerEpoch = (constants.maximumRewardRate - constants.minimumRewardRate) / totalDecayPeriod
-		return constants.maximumRewardRate - (decayRatePerEpoch * epochsAlreadyDecayed)
+	local distributionSettings = epochs.getDistributionSettings()
+	if epochIndex <= distributionSettings.rewardDecayStartEpoch then
+		return distributionSettings.minimumRewardRate
```

Comment:
Should this be maximum rate?

Reply:
And perhaps < rather than <=?

Reply:
you're right - i think this was to prevent any change to testnet, will confirm and fix and add a unit test

Reply:
this was in place for testnet to avoid changing reward ratios there - I updated the logic (it was hard for me to parse) and added some precision protections in [7a0bc29](https://github.com/ar-io/ar-io-network-process/pull/363/commits/7a0bc29e8d7a30ba3b4555d85aaf49199b31d4d3)

Reply:
tests fixed in [b78e2dc](https://github.com/ar-io/ar-io-network-process/pull/363/commits/b78e2dc7adeca45bfb0fcf5a8795ebc70585214e)